### PR TITLE
CMR-5079: Add granule validation for delete time.

### PR DESCRIPTION
**Pull request by CayvonH**

### CODE REVIEW

File: ingest-app/test/cmr/ingest/test/validation/business_rule_validation.clj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 (def variable-test-concept {:concept-type :variable})
```
Comment (CayvonH) [inline ingest-app/test/cmr/ingest/test/validation/business_rule_validation.clj]: Wondering if I should just delete this test?

File: system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj [hunk 2]
```diff
@@ -532,6 +532,22 @@
...
+                      (data-umm-c/collection 1 {}))
+          granule (-> (dg/granule-with-umm-spec-collection
+                       collection
+                       (:concept-id collection)
+                       {:data-provider-timestamps {:delete-time "2000-01-01T00:00:00Z"}})
+                      (d/item->concept :umm-json)
+                      ingest/ingest-concept)]
+      (index/wait-until-indexed)
+      (is (= 422 (:status granule)))
+      (is (contains? (set (:errors granule))
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj]: Are there other errors? I was just thinking that:
``` (is (= ["DeleteTime 2000-01-01T00:00:00.000Z is before the current time."] (:errors granule))``` is more straightforward.

File: system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj [hunk 3]
```diff
@@ -532,6 +532,22 @@
...
+    (let [collection (d/ingest-umm-spec-collection
+                      "PROV1"
+                      (data-umm-c/collection 1 {}))
+          granule (-> (dg/granule-with-umm-spec-collection
+                       collection
+                       (:concept-id collection)
+                       {:data-provider-timestamps {:delete-time "2000-01-01T00:00:00Z"}})
+                      (d/item->concept :umm-json)
+                      ingest/ingest-concept)]
+      (index/wait-until-indexed)
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj]: Remove this line.

File: system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj [hunk 4]
```diff
@@ -532,6 +532,22 @@
...
+  (testing "DeleteTime in past results in validation error"
+    (let [collection (d/ingest-umm-spec-collection
+                      "PROV1"
+                      (data-umm-c/collection 1 {}))
+          granule (-> (dg/granule-with-umm-spec-collection
+                       collection
+                       (:concept-id collection)
+                       {:data-provider-timestamps {:delete-time "2000-01-01T00:00:00Z"}})
+                      (d/item->concept :umm-json)
+                      ingest/ingest-concept)]
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj]: Separate the ingest step outside of the granule creation to make the test easier to read, and the ingest status and errors more obvious.

File: system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj [hunk 5]
```diff
@@ -532,6 +532,22 @@
...
+      (index/wait-until-indexed)
+      (is (= 422 (:status granule)))
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj]: For completeness, add a test case of ingesting a granule with DeleteTime in the future and verify that it is OK.

**Pull request merged**